plan_test.go

```diff
@@ -0,0 +1,130 @@
+package plan
+
+import (
+	"github.com/hashicorp/terraform/terraform"
+	"os"
+	"strings"
+	"testing"
+)
+
+func openReadPlan(t *testing.T, testCase string) *terraform.Plan {
+	file, err := os.Open(testCase)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	plan, err := terraform.ReadPlan(file)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	return plan
+}
+
+func TestIsDestroyedCreate(t *testing.T) {
+	isDestroyed, resourceName := IsDestroyingEBSVolume(openReadPlan(t, "create.plan"))
+
+	if exp, act := false, isDestroyed; exp != act {
+		t.Errorf("unexpected value exp=%+v\n act=%+v\n", exp, act)
+	}
+
+	if len(resourceName) != 0 {
+		t.Errorf("unexpected value")
+	}
+
+}
+
+func TestIsDestroyedTainted(t *testing.T) {
+	isDestroyed, resourceName := IsDestroyingEBSVolume(openReadPlan(t, "tainted.plan"))
+
+	if exp, act := true, isDestroyed; exp != act {
+		t.Errorf("unexpected value exp=%+v\n act=%+v\n", exp, act)
+	}
+
+	if len(resourceName) == 0 {
```

Comment:
Please be stricter in checking the result

A return of `[module.ebs.aws_ebs_volume.volume.0]` is expected for resourceNames